src/rules/boolean-keywords.coffee

```diff
@@ -0,0 +1,95 @@
+###*
+# @fileoverview This rule should require or disallow usage of specific boolean keywords.
+# @author Julian Rosse
+###
+'use strict'
+
+#------------------------------------------------------------------------------
+# Requirements
+#------------------------------------------------------------------------------
+
+TRUE_KEYWORDS = ['true', 'yes', 'on']
+FALSE_KEYWORDS = ['false', 'no', 'off']
+ALL_KEYWORDS = [...TRUE_KEYWORDS, ...FALSE_KEYWORDS]
+
+#------------------------------------------------------------------------------
+# Rule Definition
+#------------------------------------------------------------------------------
+
+module.exports =
+  meta:
+    docs:
+      description: 'enforce consistent usage of English operators'
+      category: 'Stylistic Issues'
+      recommended: no
+      # url: 'https://eslint.org/docs/rules/space-unary-ops'
+
+    schema: [
+      anyOf: [
+        type: 'object'
+        properties:
+          allow:
+            type: 'array'
+            items: [enum: ALL_KEYWORDS]
+            minItems: 2
+        additionalProperties: no
+      ,
+        type: 'object'
+        properties:
+          disallow:
+            type: 'array'
+            items: [enum: ALL_KEYWORDS]
+            minItems: 1
+        additionalProperties: no
+      ]
+    ]
+
+    messages:
+      'unexpected-fixable': "Prefer '{{ replacement }}' to '{{ unexpected }}'"
+      unexpected: "Don't use '{{ unexpected }}'"
+
+    fixable: 'code'
+
+  create: (context) ->
+    {allow, disallow} = context.options[0]
+
+    getReplacement = (trueOrFalse) ->
+      allKeywords = if trueOrFalse then TRUE_KEYWORDS else FALSE_KEYWORDS
+      allowedKeywords =
+        if allow?
+          allKeywords.filter((keyword) -> keyword in allow)
+        else
+          allKeywords.filter((keyword) -> keyword not in disallow)
+      return unless allowedKeywords.length is 1
+      [replacement] = allowedKeywords
+      replacement
+
+    #--------------------------------------------------------------------------
+    # Public
+    #--------------------------------------------------------------------------
+
+    Literal: (node) ->
```

Comment:
The `create()` method of the rule returns a visitor object whose keys are AST node type names

Notice how here we're specifying `Literal` rather than `BooleanLiteral` (which is what our AST outputs) - ESLint uses a slightly different AST format than Babel (`eslint-plugin-coffee` is converting Babel -> ESLint under the hood with the help of `babel-eslint`)